src/GetTested/Extract.hs

```diff
@@ -84,3 +90,30 @@ expandUnionVersionRange :: VersionRange -> Vector VersionRange
 expandUnionVersionRange (ThisVersion v) = Vector.singleton (ThisVersion v)
 expandUnionVersionRange (UnionVersionRanges r1 r2) = expandUnionVersionRange r1 <> expandUnionVersionRange r2
 expandUnionVersionRange _ = Vector.empty
+
+pickCompilers
+  :: Bool
+  -- ^ Enable the newest GHC version
+  -> Bool
+  -- ^ Enable the oldest GHC version
+  -> Vector Version
+  -- ^ Input
+  -> Vector Version
+pickCompilers enableNewest enableOldest versions =
+  let getNewest =
+        if enableNewest
+          then Vector.singleton $ getDown $ Vector.maximum (fmap Down versions)
```

Comment:
Not sure what I miss here, but why do we need to wrap the versions in `Down`?

Reply:
Also: `Data.Vector.maximum` is partial, so we should probably check that the vector is not empty before using it.